Sources/FBLPromises/include/FBLPromise+Retry.h

```diff
@@ -0,0 +1,130 @@
+ been exhausted. On rejection, the `work` block is reattempted after the given `delay` and will
+ continue to retry until the number of specified `attempts` have been exhausted or will bail early
+ if the given `condition` is not met.
+
+ @param count Max number of retry attempts.
+ @param interval Time to wait before the next retry attempt.
+ @param predicate Condition to check before the next retry attempt. The predicate block provides the
+                  the number of remaining retry attempts and the error that the promise was rejected
+                  with.
+ @param work A block that executes asynchronously on the default queue and returns a value or an
+             error used to resolve the promise.
+ @return A new pending promise that fulfills with the same value as the promise returned from `work`
+         block, or rejects with the same error after all retry attempts have been exhausted or if
+         the given `condition` is not met.
+ */
++ (FBLPromise *)attempts:(NSInteger)count
+                   delay:(NSTimeInterval)interval
+               condition:(nullable FBLPromiseRetryPredicateBlock)predicate
+                   retry:(FBLPromiseRetryWorkBlock)work NS_SWIFT_UNAVAILABLE("");
+
+/**
+ Creates a pending promise that fulfills with the same value as the promise returned from `work`
+ block, which executes asynchronously on the given `queue`, or rejects with the same error after all
+ retry attempts have been exhausted. On rejection, the `work` block is reattempted after the given
+ `delay` and will continue to retry until the number of specified `attempts` have been exhausted or
+ will bail early if the given `condition` is not met.
+
+ @param queue A queue to invoke the `work` block on.
+ @param count Max number of retry attempts.
+ @param interval Time to wait before the next retry attempt.
+ @param predicate Condition to check before the next retry attempt. The predicate block provides the
+                  the number of remaining retry attempts and the error that the promise was rejected
+                  with.
+ @param work A block that executes asynchronously on the given `queue` and returns a value or an
+             error used to resolve the promise.
+ @return A new pending promise that fulfills with the same value as the promise returned from `work`
+         block, or rejects with the same error after all retry attempts have been exhausted or if
+         the given `condition` is not met.
+ */
++ (FBLPromise *)onQueue:(dispatch_queue_t)queue
+               attempts:(NSInteger)count
+                  delay:(NSTimeInterval)interval
+              condition:(nullable FBLPromiseRetryPredicateBlock)predicate
+                  retry:(FBLPromiseRetryWorkBlock)work NS_REFINED_FOR_SWIFT;
+
+@end
+
+/**
+ Convenience dot-syntax wrappers for `FBLPromise` `retry` operators.
+ Usage: FBLPromise.retryOn(queue, ^id { ... })
```

Comment:
Nit: May like the simplest `retry` as an example.

Reply:
Done.